prlworkflows/sqs.py

```diff
@@ -301,18 +296,20 @@ def enumerate_sqs(structure, subl_model, endmembers=True, scale_volume=True, ski
     # error checking
     if len(subl_model) != len(structure.sublattice_model):
         raise ValueError('Passed sublattice model ({}) does not agree with the passed structure ({})'.format(subl_model, structure.sublattice_model))
-    if np.any([len(subl) for subl in subl_model] < [len(subl) for subl in structure.sublattice_model]):
-        if skip_on_failure:
-            return []
-        raise ValueError('The passed sublattice model ({}) is of lower order than the passed structure supports ({})'.format(subl_model, structure.sublattice_model))
     possible_subls = []
     for subl, abstract_subl in zip(subl_model, structure.sublattice_model):
-        if endmembers:
-            subls = itertools.product(subl, repeat=len(abstract_subl))
-        else:
-            subls = itertools.permutations(subl, r=len(abstract_subl))
+        subls = itertools.product(subl, repeat=len(abstract_subl))
         possible_subls.append(subls)
     # TODO: possibly want combinations rather than permutations because [['Cu', 'Mg']] might == [['Mg', 'Cu']]
     unique_subl_models = itertools.product(*possible_subls)
+
     # create a list of concrete structures with the generated sublattice models
-    return [structure.get_concrete_sqs(model, scale_volume) for model in unique_subl_models]
+    result = []
+    non_degenerate = []
+    for model in unique_subl_models:
+        concrete = structure.get_concrete_sqs(model, scale_volume)
+        #check for degeneracy
+        if ([concrete.sublattice_configuration,concrete.sublattice_occupancies] not in non_degenerate):
```

Comment:
1. Don't wrap the if statement in parenthesis if you don't need to 
2. Use a tuple of `(concrete.sublattice_configuration, concrete.sublattice_occupancies)` instead of a list. Tuples are immutable and among other things imply fixed length and are, which is true of your pairs of the configuration and occupancies. 
3. Since you reuse the previously mentioned tuple, use the DRY (don't repeat yourself) principle and make them a variable. 

All in all, I suggest the following:


```python
    unique_sqs = []
    unique_configurations_occupancies = []
    for model in unique_subl_models:
        proposed_sqs = structure.get_concrete_sqs(model, scale_volume)
        proposed_config_occupancy = (proposed_sqs.sublattice_configuration, proposed_sqs.sublattice_occupancies)
        #check for degeneracy
        if proposed_config_occupancy not in unique_configurations_occupancies:
            unique_configurations_occupancies.append(proposed_config_occupancy)
            unique_sqs.append(proposed_sqs)
    return unique_sqs
```

Where I also renamed the `concrete` to `propsed_sqs`.

Now if someone new (like me or like you in 6 months) comes to read this, the variable names and statements tell the whole story of what is happening:

1. For a sublattice model, we will propose a new sqs
2. That proposed sqs has some combination of a sublattice configuration and occupancies
3. If that pair of configuration and occupancies is not in our unique pairs, add it to them and add the structure itself as well
4. Go back to 1 for the next model until all are complete
5. return the unique sqs

Then you probably will find that the `#check for degeneracy` comment is redundant because from the variable names it is clear that we are checking that our new sqs is not in our set of unique ones.